src/core/algorithms/dc/model/dc.cpp

```diff
@@ -0,0 +1,104 @@
+#include "algorithms/dc/model/dc.h"
+
+#include <algorithm>
+#include <sstream>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "algorithms/dc/model/operator.h"
+#include "algorithms/dc/model/predicate.h"
+
+namespace algos::dc {
+
+bool DC::CheckAllEquality() const {
+    auto check = [](auto const& pred) {
+        return pred.GetOperator() == dc::OperatorType::kEqual and pred.IsCrossTuple() and
+               pred.IsOneColumn();
+    };
+
+    return std::all_of(predicates_.begin(), predicates_.end(), check);
+}
+
+bool DC::CheckOneInequality() const {
+    size_t count_eq = 0, count_ineq = 0;
+    for (Predicate const& pred : predicates_) {
+        if (pred.IsConstant()) return false;
+
+        Operator op = pred.GetOperator();
+        if (op == OperatorType::kEqual and pred.IsOneColumn()) {
+            count_eq++;
+        } else if (op != OperatorType::kEqual and op != OperatorType::kUnequal and
+                   pred.IsCrossTuple()) {
+            count_ineq++;
+        }
+    }
+
+    return count_eq + count_ineq == predicates_.size() && count_ineq == 1;
+}
+
+bool DC::CheckOneTuple() const {
+    ColumnOperand operand = predicates_.front().GetVariableOperand();
+    dc::Tuple tuple = operand.GetTuple();
+
+    auto check = [tuple](Predicate const& pred) {
+        return (pred.IsConstant() or pred.IsOneTuple()) and
+               pred.GetVariableOperand().GetTuple() == tuple;
+    };
+
+    return std::all_of(predicates_.begin(), predicates_.end(), check);
+}
+
+bool DC::CheckTwoTuples() const {
+    auto check = [](Predicate const& pred) { return pred.IsVariable() and pred.IsCrossTuple(); };
+
+    return std::all_of(predicates_.begin(), predicates_.end(), check);
+}
+
+DCType DC::GetType() const {
+    if (CheckAllEquality())
+        return DCType::kAllEquality;
+    else if (CheckOneInequality())
+        return DCType::kOneInequality;
+    else if (CheckOneTuple())
+        return DCType::kOneTuple;
+    else if (CheckTwoTuples())
+        return DCType::kTwoTuples;
+    else
+        return DCType::kMixed;
+}
+
+std::string DC::ToString() const {
+    if (predicates_.empty()) return {};
+
+    static constexpr char const* kNot = "!";
+    static constexpr char const* kAnd = " and ";
+
+    std::stringstream ss;
+    ss << kNot << '(' << predicates_.front().ToString();
+    for (auto pred = std::next(predicates_.begin()); pred != predicates_.end(); ++pred) {
+        ss << kAnd << pred->ToString();
+    }
```

Comment:
I would do
```suggestion
    for (Predicate pred : predicates_ | std::views::drop(1)) {
        ss << kAnd << pred->ToString();
    }
```
`std::views` is lightweight, no overhead here